kubernetes_state/datadog_checks/kubernetes_state/kubernetes_state.py

```diff
@@ -350,6 +353,14 @@ def kube_pod_status_phase(self, message, **kwargs):
                                          tags=[self._label_to_tag("pod", metric.label),
                                                self._label_to_tag("namespace", metric.label)] + self.custom_tags)
 
+            # Counts aggregated cluster-wide to avoid no-data issues on pod churn,
+            # pod granularity available in the service checks
+            tags = [
+                self._label_to_tag("namespace", metric.label),
+                self._label_to_tag("phase", metric.label)
+            ] + self.custom_tags
+            self.count(metric_name, metric.gauge.value, tags)
```

Comment:
Could you use `increment` instead of `count`? Count is for when we want to submit the count once, what we want to do here I think is increment the counter for this metric name and set of tags each time we see it.

Reply:
see https://docs.datadoghq.com/developers/metrics/counts/ for more details

Reply:
@hkaj are you sure? https://github.com/DataDog/integrations-core/blob/335436b9e5f6c59308c2f0e9526436f5f81f865a/datadog_checks_base/datadog_checks/checks/base.py#L135 seems to indicate that `increment` is deprecated.

Reply:
The message is:

```
base.py                    148 WARNING  DEPRECATION NOTICE: `AgentCheck.increment`/`AgentCheck.decrement` are deprecated, please use `AgentCheck.gauge` or `AgentCheck.count` instead, with a different metric name
```

Reply:
I guess I'm looking for some clarification from @xvello since I was copying [his PR](https://github.com/DataDog/integrations-core/pull/1277) for how to go about doing this aggregation.

Reply:
Oh, you're right, that function is not supported anymore in agent 6 :(

The issue is that calling count several time in the check doesn't work reliably for incrementation, you're supposed to call `self.count` once per run. I tested your change on my cluster and the value is wrong (it's not normalized by time).

The PR you linked actually suffers from the same problem, and is getting fixed by https://github.com/DataDog/integrations-core/pull/1840

Copying this, i.e. incrementing a `collections.Counter` and reporting its value at the end of the loop with a simple `gauge` is the way to go, according to https://github.com/DataDog/datadog-agent/blob/7820efa6b45722ce141e6af46cb0d4ef966e4e7d/pkg/aggregator/sender.go#L214-L216

Reply:
Ah excellent. Thanks for the clarification. I'll give this a go!